cli/create.go

```diff
@@ -152,35 +152,44 @@ func create(containerID, bundlePath, console, pidFilePath string, detach bool,
 	return createPIDFile(pidFilePath, process.Pid)
 }
 
-func getKernelParams(containerID string) []vc.Param {
-	return []vc.Param{
-		{
-			Key:   "init",
-			Value: "/usr/lib/systemd/systemd",
-		},
-		{
-			Key:   "systemd.unit",
-			Value: systemdUnitName,
-		},
-		{
-			Key:   "systemd.mask",
-			Value: "systemd-networkd.service",
-		},
-		{
-			Key:   "systemd.mask",
-			Value: "systemd-networkd.socket",
-		},
+var systemdParam = []vc.Param{
+	{
+		Key:   "init",
+		Value: "/usr/lib/systemd/systemd",
+	},
+	{
+		Key:   "systemd.unit",
+		Value: systemdUnitName,
+	},
+	{
+		Key:   "systemd.mask",
+		Value: "systemd-networkd.service",
+	},
+	{
+		Key:   "systemd.mask",
+		Value: "systemd-networkd.socket",
+	},
+}
+
+func getKernelParams(containerID string, needSystemd bool) []vc.Param {
+	p := []vc.Param{
 		{
 			Key:   "ip",
 			Value: fmt.Sprintf("::::::%s::off::", containerID),
 		},
 	}
+
+	if needSystemd {
+		p = append(p, systemdParam...)
+	}
+
+	return p
 }
 
 // setKernelParams adds the user-specified kernel parameters (from the
 // configuration file) to the defaults so that the former take priority.
 func setKernelParams(containerID string, runtimeConfig *oci.RuntimeConfig) error {
-	defaultKernelParams := getKernelParamsFunc(containerID)
+	defaultKernelParams := getKernelParamsFunc(containerID, runtimeConfig.HypervisorConfig.ImagePath != "")
```

Comment:
Could you please get this check done outside of the function call, for readability purpose.
We should define a function:
```go
func isInitRamFs(config vc.HypervisorConfig) bool {
        if config.ImagePath == "" && config.InitrdPath != "" {
                return true
        }

        return false
}
```